### ISSUE TITLE
Add Github Email to TU-Wien Gitlab

### ISSUE THREAD

**Issue by filo14** [ElektraInitiative/PermaplanT#1295]
For the migration to Gitlab and so that [User Contribution Mapping](https://docs.gitlab.com/ee/user/project/import/github.html#accounts-for-user-contribution-mapping) works everyone could add their primary Github Email to their TU-Wien Gitlab Account and confirm it.

In Gitlab go to Edit Profile -> Emails -> Add new email. If you did it please check the box here.

- [x] @filo14 
- [ ] @markus2330 
- [ ] @danielsteinkogler 
- [ ] @chr-schr 
- [ ] @hatchla 
- [ ] @badnames 
- [ ] @horenso 
- [ ] @andreicristian97

**Comment by hatchla** [comment]
I am not getting an email for verification.

**Comment by horenso** [comment]
I have the same issue, I added my email but no verification email...

**Comment by markus2330** [comment]
I also did not get it the confirmation email, despite clicking on "Resend ..." and it seems to be a necessity to make the [user contribution mapping](https://docs.gitlab.com/ee/user/project/import/github.html#accounts-for-user-contribution-mapping) work? I reported the problem to TU.it. It might be related to a [upTUdate - Update](https://www.it.tuwien.ac.at/tuit-home/alle-stoermeldungen/news/stoerung-uptudate-update-2) problem they have. It seems like currently TUgitLab cannot send e-mails at all.

For @yvonnemarkl and @Bushuo I created an account with the same email. So the mapping should already work. @filo14 can you test this?

@everyone: When you are already configuring GitLab, please also upload an Avatar in TUgitLab.

**Comment by filo14** [comment]
@markus2330 Thanks for looking that up 

For testing: I can't create projects in Gitlab sadly and the import from Github options (that includes issues etc.) is only from creating a project.

**Comment by markus2330** [comment]
@ElektraInitiative/permaplant sending of emails on git.tuwien.ac.at works again, please, under "Edit Profile" do:

- "Profile": set Avatar picture
- set "Emails -> Add new email" with the email address you use on GitLab **and** confirm that email
- set "Notifications -> Global notification level" to watch, so that you will get emails in future from activity
- "SSH Keys": add ssh key
- tick yourself in the top post

**Comment by markus2330** [comment]
@danielsteinkogler @chr-schr @hatchla @yvonnemarkl please check if your primary public email address (temporarily) is correctly set in GitHub, otherwise your issues/PRs might not be correctly linked. Primary email should be enough as for @andreicristian97 but we cannot check the difference if it is hidden. Please confirm if you see this.

**Comment by hatchla** [comment]
> @danielsteinkogler @chr-schr @hatchla @yvonnemarkl please check if your primary public email address (temporarily) is correctly set in GitHub, otherwise your issues/PRs might not be correctly linked. Primary email should be enough as for @andreicristian97 but we cannot check the difference if it is hidden. Please confirm if you see this.

Is this about the setting "Keep my email addresses private" in github? Should this be unticked? Or is it about adding the primary email from gitlab (that has been fetched from TU Wien SSO) in gitHUB as primary Email?

**Comment by markus2330** [comment]
> Is this about the setting "Keep my email addresses private" in github? Should this be unticked?

@hatchla Yes, it must be (temporarily) **unticked** (in Settings -> Emails).

And also "settings -> public profile -> public email" must be set as well.

> Or is it about adding the primary email from gitlab (that has been fetched from TU Wien SSO) in gitHUB as primary Email?

Yes, this is needed, too

@andreicristian97 @chr-schr @danielsteinkogler  for you it also did not work.

**Comment by hatchla** [comment]
@markus2330 This should all be done now for me.

**Comment by markus2330** [comment]
Great, thx!

@andreicristian97 @danielsteinkogler @chr-schr pls fix asap, too!

**Comment by andreicristian97** [comment]
> Great, thx!
> 
> @andreicristian97 @danielsteinkogler @chr-schr pls fix asap, too!

Done from my side

**Comment by chr-schr** [comment]
@markus2330
Should be done 👍

**Comment by markus2330** [comment]
Great, so only @danielsteinkogler left!

**Comment by markus2330** [comment]
If someone can contact him via some other channel (e.g. discord), it would be great.

**Comment by danielsteinkogler** [comment]
i just read the messages and i'm on it

**Comment by danielsteinkogler** [comment]
so my primary email in github is daniel.steinkogler@deckweiss.at and the same email adress is also verified in gitlab.
Jannis told me that he also just added his primary email of github also to gitlab

**Comment by markus2330** [comment]
Great, thx to everyone :heart: :rocket: 

I'll start the migration progress again.

Please silence here to not disturb it :no_mouth:
